### ISSUE TITLE
Can't mount mmc as mass storage on Galaxy S (i9000).

### ISSUE THREAD

**Issue by PabloPL** [xc-racer99/u-boot-aries#11]
I've moved my archlinuxarm to mmc and while i can:
- format it from linux
- mount it from linux and put files there (for example unpack archlinux there)
- boot linux from it

I'm gettting errors while trying to mount it as mass storage in uboot (and can't do anything with this)

**Comment by PabloPL** [comment]
```
[ 7282.441343] sd 1:0:0:0: [sdb] Attached SCSI removable disk
[ 7285.942136] sd 1:0:0:0: [sdb] tag#0 FAILED Result: hostbyte=DID_ERROR driverbyte=DRIVER_OK
[ 7285.942141] sd 1:0:0:0: [sdb] tag#0 CDB: Write(10) 2a 00 00 00 00 40 00 00 08 00
[ 7285.942145] print_req_error: I/O error, dev sdb, sector 64
[ 7285.942155] print_req_error: I/O error, dev sdb, sector 64
[ 7285.942159] Buffer I/O error on dev sdb1, logical block 0, lost sync page write
[ 7285.942316] EXT4-fs (sdb1): mounted filesystem with ordered data mode. Opts: (null)
[ 7286.919603] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7287.127652] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7287.335598] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7287.543655] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7287.751593] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7287.963612] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7288.072481] sd 1:0:0:0: [sdb] tag#0 FAILED Result: hostbyte=DID_ERROR driverbyte=DRIVER_OK
[ 7288.072506] sd 1:0:0:0: [sdb] tag#0 CDB: Read(10) 28 00 00 01 14 68 00 00 08 00
[ 7288.072516] print_req_error: I/O error, dev sdb, sector 70760
[ 7288.072613] EXT4-fs error (device sdb1): ext4_find_entry:1436: inode #2: comm trash.so: reading directory lblock 0
[ 7288.072657] EXT4-fs (sdb1): previous I/O error to superblock detected
[ 7288.072782] EXT4-fs warning (device sdb1): htree_dirblock_to_tree:977: inode #2: lblock 0: comm file.so: error -5 reading directory block
[ 7288.163616] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7288.375594] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7288.583614] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7288.791589] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7288.999637] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7289.207587] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7289.407587] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7289.615597] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7289.823583] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7290.031582] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7290.247640] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7290.463583] usb 1-1.2: reset high-speed USB device number 23 using ehci-pci
[ 7290.572135] sd 1:0:0:0: [sdb] tag#0 FAILED Result: hostbyte=DID_ERROR driverbyte=DRIVER_OK
[ 7290.572144] sd 1:0:0:0: [sdb] tag#0 CDB: Write(10) 2a 00 00 00 00 40 00 00 08 00
[ 7290.572148] print_req_error: I/O error, dev sdb, sector 64
[ 7290.572164] print_req_error: I/O error, dev sdb, sector 64
[ 7290.572171] Buffer I/O error on dev sdb1, logical block 0, lost sync page write
```
Logs from PC when mmc is mounted as mass storage in uboot. I've already run fsck on both partitions and it's not returning any issues.

**Comment by xc-racer99** [comment]
So, can u-boot load the uImage from MMC or are you still loading the initial uImage from SD?  Are there any logs from the u-boot side of things?

I can't really debug this as I don't have an MMC :(  The MMC support was essentially copied from Goni, one thing to check might be a different GPIO drive strength.

**Comment by PabloPL** [comment]
I'm booting linux from it (uImage and whole rootfs is on it), so maybe it's just issue in uboot config.
Will try to look at this later.

PS: Can android kernel boot without atags? I've created branch (pass-bootargs) where uboot is loading fdtfile (dtb file), so we don't have to use appended kernel zImage/uImage. It would also correctly set root device, pass all bootparams etc, so we don't have to recompile dtb after for example moving kernel from/to sdcard/onenand/sdcard to other place. It's not finished yet.

**Comment by PabloPL** [comment]
@xc-racer99 Are You using irc? If yes, i'm on linux-exynos on freenode.

**Comment by xc-racer99** [comment]
No, I'm not using IRC.  I've used a little bit in the past but not much.  I can start using it again if you think it would be useful.

Android kernel does need atags for the commandline.  Your pass-bootargs branch currently still boots Android properly.

**Comment by PabloPL** [comment]
I think that irc or anything like that would be faster than replaying using github issues ;)

**Comment by PabloPL** [comment]
@fourkbomb Could You check this on midas board? Enable usb mass storage mode and use it on internal mmc.

**Comment by PabloPL** [comment]
@xc-racer99 Fixed in https://github.com/xc-racer99/u-boot-galaxys4g/pull/20

**Comment by xc-racer99** [comment]
Merged in 7e751883b4adfe456e297a8121dbcf5a47b85c71

Thanks!
